Automattic/wp-calypso
# Checkout: display a message when related products are available for renewal

#### Changes proposed in this Pull Request
* Adds a new message to the checkout (**Composite Checkout only**) when a user has other products in their sites that are available for renewal. Depending on the quantity of products the message might change:
Many:
![Screen Shot 2020-05-14 at 17 31 42](https://user-images.githubusercontent.com/15204776/81993008-682f6f80-960a-11ea-901e-de3c72062231.png)
One:
![Screen Shot 2020-05-14 at 17 32 07](https://user-images.githubusercontent.com/15204776/81993011-69f93300-960a-11ea-837a-8181b8763cd2.png)
* It uses the UpcomingRenewalsModal implemented in #42076 to allow the user to select specific subscriptions to renew, except that the "Manage purchase" links are hidden and the primary button has a different label.
![Screen Shot 2020-05-14 at 17 31 51](https://user-images.githubusercontent.com/15204776/81993089-957c1d80-960a-11ea-85f3-01b8a8dccb9c.png)
* Depends on the `upgrades/upcoming-renewals-notices` feature flag.
* Design discussion: pbOQVh-9b-p2

#### Testing instructions

* Make sure composite-checkout is enabled for your account (You may add the flag via URL using `?flags=+composite-checkout-force,+upgrades/upcoming-renewals-notices`)
* Go to a site that has multiple purchases (at least 3) expiring soon (within the next 3 months) or that are already expired and still renewable.
* Go to Manage Purchases and open any of the purchases that are expiring soon.
* Click on renew now to go to the checkout.
* You should see the new notice telling you that there are other products also expiring.
* Click "Renew all" inside the notice to add all products to your cart.
* Remove the products from the cart, then click "other purchases" inside the notice, unselect all of them except one and click Add to cart to add only that product to the cart.
* Repeat this process until only there's one product left, notice that the message in the notice has changed to refer only to the remaining product. Click on "Add to cart" to add the product to the cart.
